app/code/Magento/CatalogGraphQl/Model/GetProductOptionFromRequest.php

```diff
@@ -0,0 +1,68 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+declare(strict_types=1);
+
+namespace Magento\CatalogGraphQl\Model;
+
+use Magento\CatalogGraphQl\Model\ProductOption\ProductOptionInterface;
+use Magento\CatalogGraphQl\Model\ProductOption\ProductOptionFactory;
+use Magento\Framework\GraphQl\Exception\GraphQlInputException;
+
+/**
+ * Class getProductOptionFromRequest
+ *
+ * Extracts product option information from request string
+ */
+class GetProductOptionFromRequest
+{
+    /**
+     * @var ProductOptionFactory
+     */
+    private $productOptionFactory;
+
+    /**
+     * getProductOptionFromRequest constructor.
+     * @param ProductOptionFactory $productOptionFactory
+     */
+    public function __construct(
+        ProductOptionFactory $productOptionFactory
+    ) {
+        $this->productOptionFactory = $productOptionFactory;
+    }
+
+    /**
+     * Extracts product option information from request string
+     *
+     * @param array $request
+     * @param string $source
+     * @return ProductOptionInterface
+     * @throws GraphQlInputException
+     */
+    public function execute(array $request, string $source): ProductOptionInterface
+    {
+        /** @var ProductOptionInterface $productOption */
+        $productOption = $this->productOptionFactory->create();
+
+        if ($source === ProductOptionInterface::SOURCE_ENTERED) {
+            $optionInfo = explode('/', base64_decode($request['id']), 2);
+            list($optionType, $optionId) = $optionInfo;
```

Comment:
Notice: Undefined offset: 1 is here
Some kind of validation could be implemented to validate `EnteredOptionInput.id` and prevent this notice.